page.go

```diff
@@ -777,46 +648,30 @@ func (page *fixedLenByteArrayPage) WriteTo(e encoding.Encoder) error {
 	return e.EncodeFixedLenByteArray(page.size, page.data)
 }
 
-func (page *fixedLenByteArrayPage) Values() ValueReader {
-	return &fixedLenByteArrayPageReader{page: page}
+func (page *fixedLenByteArrayPage) Values() PageValues {
+	return &fixedLenByteArrayValues{page: page}
 }
 
 func (page *fixedLenByteArrayPage) Buffer() BufferedPage { return page }
 
-type fixedLenByteArrayPageReader struct {
-	page   *fixedLenByteArrayPage
-	offset int
-}
-
-func (r *fixedLenByteArrayPageReader) Read(b []byte) (n int, err error) {
-	n, err = r.ReadFixedLenByteArrays(b)
-	return n * r.page.size, err
-}
+const (
+	bufferSize = 4096
```

Comment:
maybe just me but I always appreciate a note about why we chose a particular value; can help educate about what you think the constraint is that's worth optimizing for.